### PR TITLE
Refactor to include recommended serializations

### PR DESCRIPTION
**Pull request by aphillips**
This is a massive refactoring and so is best viewed via preview rather than by staring at the diff.

In this PR I brought in the text I created in localizable-manifests to describe serializations.
I then merged best practices and associated text already in this document into appropriate sections.
I split the best practices section from the introduction.
I retitled sections in best practices for accuracy.

~~In general, there is very little new/changed prose here.~~

Further edits include:
* I added a new "core" best practice, similar to my comments to e.g. VC.
* Moved additional best practices to the end of the BP section
* Retitled the section about "tech specific solutions" to just "JSON-LD" because that's all that's in there
* Moved the WebIDL BP to single-string section
* Created a U+200F image for the character reference I added (replacing a casual mention of RLM) including putting it into the echidna manifest

***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/string-meta/pull/81.html" title="Last updated on Jan 19, 2024, 7:46 PM UTC (8e0384a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/string-meta/81/f2e02c1...aphillips:8e0384a.html" title="Last updated on Jan 19, 2024, 7:46 PM UTC (8e0384a)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
 <body>
 <div id="abstract">
-  <p>This document describes the best practices for identifying language and base direction for strings used on the Web.</p>
+  <p>This document describes the best practices for identifying language and paragraph direction for strings used on the Web.</p>
```
Comment (xfq) [inline index.html:48]: Why was this changed to "paragraph direction"?

Comment (xfq) [reply index.html:48]: If we decide to use "paragraph direction", we might want to mention it in https://www.w3.org/International/articles/lang-bidi-use-cases/index.en.html#definitions too.

**Comment by aphillips** [comment]
Per 2024-01-11 call, merging this. Will make a separate PR regarding base vs. paragraph direction
